integration-libs/opf/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.component.ts

```diff
@@ -8,51 +8,51 @@ import {
   ChangeDetectionStrategy,
   Component,
   Input,
-  OnDestroy,
   OnInit,
 } from '@angular/core';
 import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
 import { OpfPaymentMethodType, PaymentSessionData } from '@spartacus/opf/root';
-import { Observable, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
 import { OpfCheckoutPaymentWrapperService } from './opf-checkout-payment-wrapper.service';
 
 @Component({
   selector: 'cx-opf-checkout-payment-wrapper',
   templateUrl: './opf-checkout-payment-wrapper.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class OpfCheckoutPaymentWrapperComponent implements OnInit, OnDestroy {
-  protected subscription = new Subscription();
+export class OpfCheckoutPaymentWrapperComponent implements OnInit {
+  @Input() selectedPaymentId: number;
 
   renderPaymentMethodEvent$ = this.service.getRenderPaymentMethodEvent();
+  isInitPaymentFailed$ = this.service.getIsInitPaymentFailed();
```

Comment:
Instead of creating additional behaviour subject in the service. What do you think about extending responsibility of `renderPaymentMethodEvent$`?

I guess just `isError` property has to be added into `OpfRenderPaymentMethodEvent` interface.

```
export interface OpfRenderPaymentMethodEvent {
  isLoading: boolean;
  isError: boolean;
  renderType?: OpfPaymentMethodType | null;
  data?: string | null;
}
```